prow/cmd/webhook-server/helpers.go

```diff
@@ -144,25 +147,25 @@ func isCertValid(cert string) error {
 	return nil
 }
 
-func createSecret(client ClientInterface, ctx context.Context, expiry int, dns []string) (string, string, string, error) {
-	if err := client.CreateSecret(ctx, secretID); err != nil {
+func createSecret(client ClientInterface, ctx context.Context, o options) (string, string, string, error) {
+	if err := client.CreateSecret(ctx, o.secretID); err != nil {
 		return "", "", "", fmt.Errorf("unable to create secret %v", err)
 	}
 
-	serverCertPerm, serverPrivKey, caPem, err := updateSecret(client, ctx, expiry, dns)
+	serverCertPerm, serverPrivKey, caPem, err := updateSecret(client, ctx, o)
 	if err != nil {
 		return "", "", "", fmt.Errorf("unable to write secret value %v", err)
 	}
 	return serverCertPerm, serverPrivKey, caPem, nil
 }
 
-func updateSecret(client ClientInterface, ctx context.Context, expiry int, dns []string) (string, string, string, error) {
-	serverCertPerm, serverPrivKey, caPem, secretData, err := genSecretData(expiry, dns)
+func updateSecret(client ClientInterface, ctx context.Context, o options) (string, string, string, error) {
```

Comment:
doesn't sound like the best idea changing the signature to require all options, might be better to be more explicit on what this function requires